### PR TITLE
Support QL 'trim'

### PR DESCRIPTION
**Pull request by qcorporation**
Jira Title:
Support QL 'trim'
https://softwaresecured.atlassian.net/browse/OM-1238

Description:
create a semantic unit test for the new functorOp 'trim'
added functorOp definition
handled the InterpreterEngine new functorOp
added synthesiser implementation for the trim
added trim into scanner

Test Cases,
ran string_trim.dl and validated that the output from the test is expected (found in A.csv)

### CODE REVIEW

File: src/InterpreterEngine.cpp [hunk 1]
```diff
@@ -337,6 +338,18 @@ RamDomain InterpreterEngine::execute(const InterpreterNode* node, InterpreterCon
...
 
+                case FunctorOp::TRIM: {
+                    auto symbol = execute(node->getChild(0), ctxt);
+                    const std::string& str = getSymbolTable().resolve(symbol);
+                    std::string trim_str;
+                    try {
+                        trim_str = profile::Tools::trimWhitespace(str);
+                    } catch (...) {
+                        std::cerr << "warning: trim failed from value ( \"" << str << "\" ) functor.";
+                    }
```
Comment (ohamel-softwaresecure) [inline src/InterpreterEngine.cpp]: `trim` shouldn't throw. :(

Comment (qcorporation) [reply src/InterpreterEngine.cpp]: I know we talked about this but this is why I did it:

The problem is:
profile::Tools::trimWhitespace->str.erase gives a prototype
@throw std::out_of_range If @a pos is beyond the end of this string.

Comment (ohamel-softwaresecure) [reply src/InterpreterEngine.cpp]: Yes, but the logic of the impl should never issue such an erase call.

Comment (qcorporation) [reply src/InterpreterEngine.cpp]: you are right, i was worried about __str.size() getting mangled, but it seems that _M_check checked the same size() to throw out of range, I'll remove

**Pull request merged**